repositories/EventRepository.ts

```diff
@@ -40,9 +40,22 @@ export class EventRepository extends BaseRepository<EventModel> {
     return this.repository.remove(event);
   }
 
-  public async isUnusedAttendanceCode(attendanceCode: string): Promise<boolean> {
-    const count = await this.repository.count({ attendanceCode });
-    return count === 0;
+  public async isAvailableAttendanceCode(attendanceCode: string, start: Date, end: Date): Promise<boolean> {
+    const startsDuringCurrentEvent = await this.repository.find({
+      where: {
+        attendanceCode,
+        start: LessThanOrEqual(end) && MoreThanOrEqual(start),
+      },
+    });
+
+    const endsDuringCurrentEvent = await this.repository.find({
+      where: {
+        attendanceCode,
+        end: LessThanOrEqual(end) && MoreThanOrEqual(start),
+      },
+    });
+
+    return (startsDuringCurrentEvent.length === 0) && (endsDuringCurrentEvent.length === 0);
```

Comment:
I think this could be optimized into just one query. Can you double check if typeorm has an `OR` function in queries? I think you can just specify multiple `where` conditions